File: pinpointB/src/main/java/com/nxquar/pinpoint/service/NoteService.java
package com.nxquar.pinpoint.service;

import com.nxquar.pinpoint.DTO.MessageResponse;
import com.nxquar.pinpoint.Model.Note;
import org.aspectj.weaver.ast.Not;

import java.util.List;
import java.util.UUID;

public interface NoteService {
   Note createNote(Note note, String jwt);
   MessageResponse UpdateNote(Note note, String jwt);
   MessageResponse DeleteNote(UUID noteId, String jwt);
   List<Note> getAllNotes(String jwt);
   Note getNoteById(UUID id, String jwt);

}

File: pinpoint/lib/data/datasource/notes/note_service.dart
import 'package:dio/dio.dart';
import 'package:pinpoint/model/notes/notes.dart';
import 'package:pinpoint/resources/constant/string/appString.dart';
import 'package:retrofit/retrofit.dart';

part 'note_service.g.dart';

@RestApi(baseUrl: "${AppString.baseUrl}")
abstract class NoteService {
  factory NoteService(Dio dio, {String baseUrl}) = _NoteService;

  @GET("/notes")
  Future<List<Note>> getAllNotes(
  );

  @GET("/notes/{id}")
  Future<Note> getNoteById(
    @Path("id") String id,
  );

  @POST("/notes")
  Future<Note> createNote(
    @Body() Note note,
  );

  @PUT("/notes")
  Future<HttpResponse> updateNote(
    @Body() Note note,
  );

  @DELETE("/notes/{id}")
  Future<HttpResponse> deleteNote(
    @Path("id") String id,
  );
}
